### ISSUE TITLE
Yoda comparisons not flagged against constants

### ISSUE THREAD

**Issue by paulschreiber** [WordPress-Coding-Standards/WordPress-Coding-Standards#290]
The lack of Yoda comparisons not flagged against constants, only against strings

```php
    if ( $taxonomy === MyClass::TAXONOMY_SLUG ) { // not flagged
        $link = true;
    } elseif ( $taxonomy === 'post_tag' ) { // flagged
        $link = false;
    }
```

If you correct the flagged errors, you end up with:

```php
    if ( $taxonomy === MyClass::TAXONOMY_SLUG ) { 
        $link = true;
    } elseif (  'post_tag' === $taxonomy ) { // flagged
        $link = false;
    }
```

which is inconsistent, and looks like a bug.
